tools/docs/generate2.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""A tool to generate api_docs for TensorFlow Addons.
+
+```
+Run from root folder of source code
+python tools/docs/generate2.py --output_dir=docs/
+```
+
+Requires a local installation of:
+  Docs Generator - pip install git+https://github.com/tensorflow/docs
+  tensorflow_addons
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from os import path
+
+from absl import app
+from absl import flags
+import tensorflow_addons as tfa
+
+from tensorflow_docs.api_generator import doc_controls
+from tensorflow_docs.api_generator import generate_lib
+from tensorflow_docs.api_generator import parser
+
+from tensorflow.python.util import tf_inspect
+
+# Use tensorflow's `tf_inspect`, which is aware of `tf_decorator`.
+parser.tf_inspect = tf_inspect
```

Comment:
Sean, it seems that tensorflow_docs cannot recognize those imported modules/function, such as keras_utils, absolute_import etc. Not sure whether that's one reason to use tf_export in tensorflow.

<img width="600" alt="Screen Shot 2019-04-11 at 4 48 48 PM" src="https://user-images.githubusercontent.com/1112263/55943725-b3de4f80-5c79-11e9-8395-442034c231aa.png">

Reply:
@markdaoust

Reply:
It will be good to find out a way to link them back to the documentation in TF core.

Reply:
> Can we confirm if this is required? I'm not aware of us using tf_decorator, but it can probably be empirically checked.

Correct, this line only does anything for you if you're using tf_decorator.

> it seems that tensorflow_docs cannot recognize those imported modules/function, such as keras_utils, absolute_import etc. Not sure whether that's one reason to use tf_export in tensorflow.

I'm not sure I understand the question here what is the problem, what are we trying to acomplish?.
I should block the generator from ever mentioning "absolute_import", "division", "print_function".

Note that one of the functions of the base_dir argument is to allow it to filter out objects which are imported by your package, but not part of your package. No docs are generated for objects defined in files outside of base_dir.